// react-fast-highlight/lib/esm/Highlight.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import * as React from 'react';
import PropTypes from 'prop-types';
import hljs from 'highlight.js';
import BareHighlight from './BareHighlight';

var Highlight = function Highlight(props) {
  var children = props.children,
      rest = _objectWithoutPropertiesLoose(props, ["children"]);

  return /*#__PURE__*/React.createElement(BareHighlight, Object.assign({}, rest, {
    highlightjs: hljs
  }), children);
};

Highlight.defaultProps = {
  className: '',
  languages: [],
  worker: null
};
process.env.NODE_ENV !== "production" ? Highlight.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
  languages: PropTypes.arrayOf(PropTypes.string),
  worker: PropTypes.object // eslint-disable-line react/forbid-prop-types

} : void 0;
export default Highlight;
